src/containers/Banner.tsx

```diff
@@ -262,12 +308,15 @@ const Banner = ({ isEditBanner, user }: BannerProps) => {
         {user?._id === authUser._id ? (
           !isEditBanner && <EditProfileForm />
         ) : (
-          <Button
+          <StyledButton
             className="rounded-pill fw-bold px-2 mt-3 me-3"
             variant="dark"
+            onClick={handleFollowClick}
+            onMouseOver={handleOnMouseOver}
+            onMouseLeave={handleOnMouseLeave}
           >
-            {t("common.follow")}
-          </Button>
+            {buttonText}
```

Comment:
add spinner instead of text if `followedUser.isLoading` and disable button to prevent additional press